library/core/src/slice/mod.rs

```diff
@@ -1099,10 +1099,13 @@ impl<T> [T] {
     /// assert!(iter.next().is_none());
     /// ```
     ///
-    /// There's no `windows_mut`, as that existing would let safe code violate the
-    /// "only one `&mut` at a time to the same thing" rule.  However, you can sometimes
-    /// use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
+    /// There's no `windows_mut` Iterator-equivalent of `windows`,
+    /// (as `[0,1,2].windows_mut(2).collect()` would create 2 mutable references to the middle elements of that array),
```

Comment:
Instead of an example, I think it may be worth giving the reasoning here. For example, "... because the `Iterator` trait cannot represent the required lifetimes (though this is possible with a [`LendingIterator`]). However, you can sometimes..."

Reply:
I've now lead with this more abstract lifetime reasoning, though I did not cut the counter(!)example, as I think it is a simple to understand but powerful proof of non-existence of windows_mut as an Iterator. Let me know what you think!